From b077575ee509d72dfa2f88224778cb7ad7ef65a9 Mon Sep 17 00:00:00 2001
From: nickcooper-zhangtonghao <nic@opencloud.tech>
Date: Fri, 3 Mar 2017 01:37:21 -0800
Subject: [PATCH] mcast-snooping: Avoid segfault for vswitchd.

The ports which are attached mrouters or hosts, were destroyed
by users via ovs-vsctl commands. Currently the vswitch will
segfault if users use "ovs-appctl mdb/show" to show mdb info.
This patch avoids a segfault.

Signed-off-by: nickcooper-zhangtonghao <nic@opencloud.tech>
Signed-off-by: Ben Pfaff <blp@ovn.org>
diff --git a/ofproto/ofproto-dpif.c b/ofproto/ofproto-dpif.c
index 888babb..759c9f3 100644
--- a/ofproto/ofproto-dpif.c
+++ b/ofproto/ofproto-dpif.c
@@ -2833,6 +2833,7 @@ bundle_destroy(struct ofbundle *bundle)
     }
 
     bundle_flush_macs(bundle, true);
+    mcast_snooping_flush_bundle(ofproto->ms, bundle);
     hmap_remove(&ofproto->bundles, &bundle->hmap_node);
     free(bundle->name);
     free(bundle->trunks);
